running-code/execute-script.md

```diff
@@ -0,0 +1,191 @@
+which, in addition to allowing specifying the version of Python to run, can also read shebang lines and emulate some of that behavior.
+This allows for shebang lines to be re-used between Linux, macOS, and Windows systems. However, on Windows the command must still
+be prefaced with another command (`py`).
+
+```bash
+py my_program.py
+# ✨ Hello from Python ✨
+```
+
+:::{tip}
+While all Python files should end in a `.py`, this naming is necessary for Windows to associate a script with Python, as opposed
+to Linux where `.py` is a convention and the shebang associates the file with Python.
+
+Also, although there is no in-source format that can tell Windows what to do with a Python file, executing a
+Python file with a shebang on Windows also does not cause any issues. Python just sees the whole line as
+a comment and ignores it!
+
+Because of these differences it is best practice to use both a shebang and `.py` for all Python scripts.
+:::
+
+## Executable comparisons
+
+### Pros of passing a file to `python`:
+- don't need execute permissions
+- works on every system
+- explicit about what you expect to happen
+
+### Pros of inserting a shebang to the file:
+- file is associated with specific python
+  - don't have to remember which
+- don't have to use the `python` command
+  - don't have to even remember it is a Python script
+
+
+## Separating script from import behavior
+
+Sometimes a Python file that is useful to execute is also useful to import. You may want to use `shiny_hello`
+in another Python file. But right now, the `my_program.py` does all its script behavior even when it is imported. Consider
+
+```python
+import my_program
+
+def guess_my_number():
+    my_program.shiny_hello()
+    print("Was your number 42?")
+
+guess_my_number()
+# ✨ Hello from Python ✨
+# ✨ Hello from Python ✨
+# Was your number 42?
+```
+
+You may not have expected it to print the hello twice, but it did. This is because `my_program` is set to
+_always_ call `shiny_hello`, and now `guess_my_number` also calls it. That's two times. How can we make
+`my_program` only call `shiny_hello` when it is used as a script?
+
+You may have already seen the answer, without realizing what it was doing. `my_program` needs a conditional that checks if is is in "script mode" or "import mode" and that conditional is `if __name__ == "__main__":`.
+
+This conditional is often used at the bottom of modules, especially modules that are expected to be executed
+directly, to separate code that is intended to execute as part of a command from code that is intended to
+execute as part of an import.
+
+```python
+#!/usr/bin/env python
+# The above line is a shebang, and can take the place of typing python on the command line
+# This comment is below, because shebangs must be the first line!
+
+def shiny_hello():
+    print("\N{Sparkles} Hello from Python \N{Sparkles}")
+
+
+if __name__ == "__main__":
+    shiny_hello()
+```
```

Comment:
````suggestion
```python
#!/usr/bin/env python
# The above line is a shebang and can take the place of typing python on the command line
# This comment is below, because shebangs must be the first line!

def shiny_hello():
    print("\N{Sparkles} Hello from Python \N{Sparkles}")

# By calling shiny_hello within this conditional, it will only run when this file is run as a script. It will 
# not run if you import it into another script. 
if __name__ == "__main__":
    shiny_hello()
```
````